label-owners: Add Arrikto folk to relevant labels

yanniszark opened the pull request:
This commit supplements the label-owners list by adding Arrikto folk to
their relevant labels.

It also adds the area/security label, which isn't in the label-owners
file yet.

/cc @jlewi @vkoukis @jbottum 

Signed-off-by: Yannis Zarkadas <yanniszark@arrikto.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/community/311)
<!-- Reviewable:end -->

jlewi commented on labels-owners.yaml:
```diff
@@ -35,6 +36,7 @@ labels:
     owners:
       - animeshsingh
       - cliveseldon
+      - cvenets
```
@yanniszark what are you and @cvenets working on with respect to inference? I think inference is mostly kfserving.

yanniszark replied:
@cvenets has been working on PVC support for KFServing.
However, discussions have been made mainly in community meetings and not GitHub.
I'm removing him for now for this PR to go forward.

jlewi commented:
We have an area/enterprise_readiness label. Is it also worth adding the area/security label? How would the two differ?

yanniszark commented:
@jlewi I have updated the addressing your suggestions.
I have added the `area/enterprise_readiness` label which wasn't included, as suggested.
Can you please take another look?

jlewi commented on labels-owners.yaml:
```diff
@@ -82,6 +88,7 @@ labels:
   area/pipelines:
     owners:
       - ark-kun
+      - elikatsis
```
What's the reason for adding @elikatsis here? Is @elikatsis in any of the pipelines OWNERs files or able to speak to pipelines ROADMAP?

The criterion I'm following is who should people ask to identify whether issue X will be fixed. 

@jessiezcc seems like the person driving the roadmap for pipelines.

yanniszark replied:
@elikatsis is driving Arrikto's contributions to Kubeflow Pipelines and has delivered a complete design and implementation of PersistentVolume/Snapshot support in the Pipelines SDK.
https://github.com/kubeflow/pipelines/issues/801
https://github.com/kubeflow/pipelines/pull/926

Other than that, he has been actively helping out in closing bugs, as can be seen from the regular stream of PRs:
https://github.com/kubeflow/pipelines/pulls?q=is%3Apr+author%3Aelikatsis+is%3Aclosed

Totally, he has contributed more than 5000 loc.

I believe this is more than enough to be included here.
@elikatsis is the owner of the code for PVC/Snapshot support in Kubeflow Pipelines.

jlewi replied:
I don't doubt that @elikatsis has made significant and impactful contributions to pipelines.

Perhaps we should discuss at a community meeting?

To me the point of listing owner is who is accountable and responsible for maintaing each area going forward. One of the key criterion is knowing who is the owner responsible for every sub-piece in an area (eg. DSL, data versioning, UI, metadata, TFX integration, etc...) so that they can route issues.

My expectation is that right now @jessiezcc how the most visibility and context into KFP as a whole.

For example, if a user reports a problem with the DSL who should I assign that bug to verify if it is in fact  abug, assess priority, and assign someone to fix it?

My sense is that @jessiezcc is the best person for that because she has the most visibility into pipelines as a whole.

My expectation would be that she would have the context to know which areas of KFP are owned/maintained by Arrikto and reassign issues as necessary.

When it comes to KFP, I would defer to @jessiezcc and the other KFP owners to decide who should be listed here. So if you'd like @elikatsis please raise it with here and get her to LGTM it.

jlewi replied:
To elaborate a bit more. The immediate goal of identifying area owners is to identify who can and is responsible for issue triage.

There are currently 236 issues waiting on triage by area owners
https://github.com/orgs/kubeflow/projects/26

Pipelines is under-represented because we don't have the Triage GitHub action running on the pipelines repo.

yanniszark commented:
@jlewi I updated the PR with your proposed changes, we'll tackle the pipelines issue in another PR.
Does it look good now?

jlewi commented on labels-owners.yaml:
```diff
@@ -44,8 +49,11 @@ labels:
       - krishnadurai
       - kunmingg
       - lluunn
+      - yanniszark
```
How come your adding yourself as an owner of centraldashboard? Are you actually working on centraldashboard?

yanniszark replied:
I haven't, I think you're not viewing the diff correctly.
This is the area/istio label.

jlewi commented on labels-owners.yaml:
```diff
@@ -44,8 +49,11 @@ labels:
   area/jupyter:
     owners:
+      - elikatsis
+      - ioandr
```
@ioandr and @elikatsis  don't seem that active with respect to Jupyter. Do they really have the context to triage and help out on jupyter related issues?

https://github.com/pulls?q=+is%3Apr+author%3Aioandr+org%3Akubeflow+

https://github.com/pulls?q=is%3Aissue+author%3Aelikatsis+org%3Akubeflow+

yanniszark replied:
@ioandr made the original contribution of the Jupyter Web App and @elikatsis has worked on issues we had with the JWA.
My first understanding of label-owners was to add the people who are knowledgeable / own a piece of the code. 
From your comments I understand that it's meant to be used as a list of people active in day-to-day issues, not so much as a record of who knows what.
In that case, I will remove those entries.

jlewi commented:
@yanniszark I made a couple suggestions. My general feeling is that in a PR you should only be adding yourself and not adding others.

Adding oneself as a label owner should be an acknowledgment that you accept certain responsibilities; e.g. triaging all the issues in that area.

So how about scoping this PR to just adding yourself to the labels you want to take on?

jlewi commented:
/lgtm
/approve
